e2e/test/iothub/service/QueryClientE2ETests.cs

```diff
@@ -18,6 +18,7 @@ namespace Microsoft.Azure.Devices.E2ETests.IotHub.Service
     [TestClass]
     [TestCategory("E2E")]
     [TestCategory("IoTHub")]
+    [DoNotParallelize] // creating jobs limits to running one at a time anyway, so avoid throttling conflicts
```

Comment:
These tests may be run on different .NET frameworks simultaneously, so this doesn't protect us completely from throttling

Reply:
Not completely, but we can at least reduce it.

Reply:
If we make other changes here, we could probably remove this constraint.

Maybe class initialize queries to see if at least one job is there, and if not, creates and immediately cancels one, for example.

Reply:
This set of tests are for querying devices, the problem likely from other test that is creating device or removing devices that is running in parallel.

There is a idPrefix string that may add uniqueness to the runs. What about we add the host-info (like hostname, or agentId etc) to the prefix so parallel run on different agent have its own scope of devices prefix?

Reply:
> This set of tests are for querying devices, the problem likely from other test that is creating devices or removing devices in parallel.

Tests shouldn't be removing devices with Ids they didn't create, so I don't anticipate that being a problem.